base/strings/types.jl

```diff
@@ -19,10 +19,10 @@ struct SubString{T<:AbstractString} <: AbstractString
     function SubString{T}(s::T, i::Int, j::Int) where T<:AbstractString
         i > j && return new(s, 0, 0) # allow i > j as it is consistent with getindex
         i < 1 && throw(BoundsError(s, i))
-        j > endof(s) && throw(BoundsError(s, j))
+        j > s.len && throw(BoundsError(s, j))
```

Comment:
it's not guaranteed that all AbstractString subtypes will have a `.len` field - that's even going to be removed from String soon